p2p/src/service_impl/libp2p/mod.rs

```diff
@@ -676,7 +677,16 @@ impl Libp2pService {
                                     addresses
                                         .iter()
                                         .filter_map(|a| {
-                                            P2pConnectionOutgoingInitLibp2pOpts::try_from(a).ok()
+                                            // `a` misses the p2p part
```

Comment:
I would expand on this, does something like this make sense @akoptelov ?

```rust
// Extend the multiaddress `a` by appending the p2p protocol with the peer ID.
// This is necessary because `a` initially lacks the p2p component required for a complete address.
```

Reply:
Maybe it deserves its own function
`fn extend_to_complete_address(addr: UncompleteMultiaddress) -> CompleteP2PAddress {..}`